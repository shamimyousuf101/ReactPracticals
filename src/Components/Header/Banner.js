import React from 'react';

const Banner = ({subHeading}) =>  (<header className="banner"><h1>Promotion Manager</h1><h3>{subHeading}</h3></header>);
export default Banner;

