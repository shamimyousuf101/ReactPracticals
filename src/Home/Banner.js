import React from 'react';

const Banner = ({ subHeading }) => {
    return (<div className="banner"><h1>Promotion Manager</h1><h3>{subHeading}</h3></div>)
}

export default Banner;
