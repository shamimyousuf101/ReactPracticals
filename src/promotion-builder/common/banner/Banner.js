import React from 'react';
import PropsTypes from 'prop-types';

import './banner.css';
import block from 'bem-cn';
const b = block('banner');

const Banner = ({subHeading}) =>  (<header className={b()}><h1>Promotion Manager</h1><h3>{subHeading}</h3></header>);

Banner.propTypes = {
    subHeading: PropsTypes.string.isRequired
}

export default Banner;

