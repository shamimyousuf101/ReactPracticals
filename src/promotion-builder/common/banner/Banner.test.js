import React from "react";
import { shallow } from "enzyme";

import Banner from "./Banner";

describe("<Banner>", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Banner subHeading={"Test"} />);
  });

  it("renders with correct class name", () => {
    expect(wrapper.hasClass("banner")).toBe(true);
  });

  describe("Props", () => {
    it("renders with correct prop value", () => {
      expect(
        wrapper
          .children()
          .find("h3")
          .props().children
      ).toBe("Test");
    });
  });
});
