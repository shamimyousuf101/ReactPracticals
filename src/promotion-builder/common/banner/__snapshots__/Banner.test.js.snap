// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner component renders correctly 1`] = `
<header
  className="banner"
>
  <h1>
    Promotion Manager
  </h1>
  <h3>
    Test
  </h3>
</header>
`;
